superset-frontend/src/features/databases/UploadDataModel/index.tsx

```diff
@@ -329,6 +337,18 @@ const UploadDataModal: FunctionComponent<UploadDataModalProps> = ({
     form.resetFields();
   };
 
+  const validateTableName =
+    (currentDatabase: CurrentDatabase) => (_: any, value: string) => {
+      const isPrestoTrino = currentDatabase?.label
+        ?.toLowerCase()
+        .includes('trino');
+
+      if (isPrestoTrino && value.includes(' ')) {
+        return Promise.reject(t('Trino table names must not contain spaces'));
+      }
+      return Promise.resolve();
```

Comment:
It doesn't seem ideal to explicitly check for specific cases. Instead, I think it's better to catch this issue during the backend process that executes the 'CREATE TABLE' statement and deliver it as an error message.